### PR TITLE
Merge Branch-DOB-TestSuite Hong Liang with master

### LINKED ISSUE #137: Improve test suite for AddDOB feature

### PR DESCRIPTION
**Pull request by tanhl2000**
closes #137

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 1]
```diff
@@ -132,4 +124,22 @@ public void toModelType_nullGender_throwsIllegalValueException() {
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Gender.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
     }
+
+    @Test
+    public void toModelType_invalidDob_throwsIllegalValueException() {
+        JsonAdaptedPerson person =
+                new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, INVALID_EMAIL,
+                        VALID_ADDRESS, VALID_GENDER, INVALID_DOB, VALID_UID);
+        String expectedMessage = Email.MESSAGE_CONSTRAINTS;
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java]:
```suggestion
        String expectedMessage = Date.MESSAGE_CONSTRAINTS;
```

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -74,10 +75,10 @@ public class CommandTestUtil {
     static {
         DESC_AMY = new EditPersonDescriptorBuilder().withName(VALID_NAME_AMY)
                 .withPhone(VALID_PHONE_AMY).withEmail(VALID_EMAIL_AMY).withAddress(VALID_ADDRESS_AMY)
-                .withGender(VALID_GENDER_AMY).build();
+                .withDob(VALID_DOB_AMY).withGender(VALID_GENDER_AMY).build();
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: I think the dob field can be moved after gender for parameter order consistency
```suggestion
                .withGender(VALID_GENDER_AMY).withDob(VALID_DOB_AMY).build();
```

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 3]
```diff
@@ -74,10 +75,10 @@ public class CommandTestUtil {
...
         DESC_BOB = new EditPersonDescriptorBuilder().withName(VALID_NAME_BOB)
                 .withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB)
-                .withGender(VALID_GENDER_BOB).build();
+                .withDob(VALID_DOB_BOB).withGender(VALID_GENDER_BOB).build();
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: Same as above
```suggestion
                .withGender(VALID_GENDER_BOB).withDob(VALID_DOB_BOB).build();
```

File: src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java [hunk 4]
```diff
@@ -51,6 +52,10 @@ public void equals() {
         editedAmy = new EditPersonDescriptorBuilder(DESC_AMY).withAddress(VALID_ADDRESS_BOB).build();
         assertFalse(DESC_AMY.equals(editedAmy));
 
+        // different dob -> returns false
+        editedAmy = new EditPersonDescriptorBuilder(DESC_AMY).withDob(VALID_DOB_BOB).build();
+        assertFalse(DESC_AMY.equals(editedAmy));
+
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java]: Same as above, it might be better to move these 3 lines below gender

**Review by Qiaoran-M — CHANGES_REQUESTED** [review]
LGTM except for minor changes!

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 5]
```diff
@@ -132,4 +124,22 @@ public void toModelType_nullGender_throwsIllegalValueException() {
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Gender.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
     }
+
+    @Test
+    public void toModelType_invalidDob_throwsIllegalValueException() {
+        JsonAdaptedPerson person =
+                new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, INVALID_EMAIL,
```
Comment (Vinodjayakumar124) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java]:
```suggestion
                new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL,
```

**Review by Vinodjayakumar124 — CHANGES_REQUESTED** [review]
Just 1 tiny error I spotted! Looks good otherwise

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java (hunk 1 above)
Comment (tanhl2000) [reply src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java]: ahhh ok will change

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 2 above)
Comment (tanhl2000) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: thx for catching!

**Review by Qiaoran-M — APPROVED** [review]
LGTM!

**Review by Vinodjayakumar124 — APPROVED** [review]

**Pull request merged**